### PR TITLE
Fix 404s on Codecov File Views

### LINKED ISSUE #203: Apply Path Fixing for Codecov
**Description**:
When code coverage reports are uploaded, they don't match the repo path. This means you cannot view the files within codecov (e.g. clicking on a file in the following URL will result in a 404, https://codecov.io/gh/ral-facilities/datagateway-api/tree/master/datagateway-api/datagateway_api/common). Path fixing is the solution to this, some information about it here - https://docs.codecov.io/docs/fixing-paths

**Acceptance criteria**:
- [ ] You should be able to view a copy of each file, displaying code coverage data (e.g. https://codecov.io/gh/ral-facilities/scigateway/src/master/src/contactPage/contactPage.component.tsx). For master, this might not be testable until the PR is merged, but I'll change the main branch on codecov so hopefully this can be tested before merging
- [ ] GitHub Actions build should still pass
- [ ] Codecov PR comment should be present and correct

### PR DESCRIPTION
**Pull request by MRichards99**
This PR will close #203 

## Description
As per the linked issue, there was a bug regarding Codecov and accessing the API's files within the codecov portal to see the colourful display of test coverage (screenshot of the issue below). By not providing a path for the checkout action within the CI build workflow, this maps the path on the runner to the relative path on the repo. This is a slightly different solution to the 'path fixing' I identified in the issue, but changing the repo path on the CI runner has turned out to be the simpler solution. This means you can see coverage data in a colourful format!

image

## Testing Instructions
Viewing any of the files using the following link (codecov on this branch) will allow you to view them with coverage data:
https://codecov.io/gh/ral-facilities/datagateway-api/tree/bugfix%2Fcodecov-path-fixing/datagateway_api

- [x] Review code
- [x] Check GitHub Actions build
- [x] Review changes to test coverage
- [x] You should be able to view a copy of each file, displaying code coverage data - test link provided above
- [x] Codecov PR comment should be present and correct on this PR

## Agile Board Tracking
Connect to #203

### CODE REVIEW

**Review by patrick-austin — APPROVED** [review]
Didn't find any 404s, but there's still some weirdness with Codecov. For example:

- Navigating to Codecov through the PR comment (rather than using the "Pulls" tab on the Codecov website) doesn't show anything: https://codecov.io/gh/ral-facilities/datagateway-api/pull/206?src=pr&el=h1 VS https://app.codecov.io/gh/ral-facilities/datagateway-api/compare/206
- Navigating through the infographic (sunburst) takes you to `tree` rather than `src`, which again doesn't display anything: https://codecov.io/gh/ral-facilities/datagateway-api/tree/bugfix%2Fcodecov-path-fixing/datagateway_api/common/database/filters.py VS https://codecov.io/gh/ral-facilities/datagateway-api/src/bugfix%2Fcodecov-path-fixing/datagateway_api/common/database/filters.py

But given that these aren't the 404 issue, and I've seen them before in other repos I don't think it's related to this issue so happy to approve this as long as you weren't expecting this to address the above points.

**Comment by MRichards99** [comment]
Yeah I can recreate the issues you're describing, not sure what the solution would be for them though. I wonder if the first one is occurring because the base branch (master) doesn't have the changes in, so there's nothing to compare to. Perhaps that's more hopeful than realistic though. I will dig a bit deeper into the sunburst issue once I've merged this

**Comment by MRichards99** [comment]
It looks like the sunburst issue has been resolved - if you navigate to a file (or directory for that matter) it'll take you to `src` and displays files as intended. I guess we can find out if the first issue has been resolved when a PR is opened that contains the changes from this PR

**Pull request merged**
